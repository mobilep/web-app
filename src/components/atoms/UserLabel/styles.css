.UserLabel {
	display: flex;
	align-items: center;
	padding: 6px 23px 6px 6px;
	border-radius: 100px;
	background-color: var(--grey1);
}

.UserLabel-avatar {
	margin-right: 7px;
}
