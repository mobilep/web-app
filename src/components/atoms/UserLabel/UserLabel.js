import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import { Avatar } from '..';

import './styles.css';

const getClassName = (className) => classNames('UserLabel', className);

const UserLabel = ({ fullName, initials, imgUrl, className }) => (
	<div className={getClassName(className)}>
		<Avatar
			className='UserLabel-avatar'
			initials={initials}
			img={imgUrl}
			size={Avatar.AVATAR_SIZES.AVATAR_22}
		/>

		{fullName}
	</div>
);

UserLabel.propTypes = {
	className: PropTypes.string,
	fullName: PropTypes.string,
	imgUrl: PropTypes.string,
	initials: PropTypes.string,
};

export default UserLabel;
