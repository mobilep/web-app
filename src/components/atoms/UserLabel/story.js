import React from 'react';
import { text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';

import UserLabel from './UserLabel';

storiesOf('1. Atoms / UserLabel', module)
	.add('UserLabel', () => <UserLabel initials={text('initials', 'JT')} fullName={text('fullName', 'Jane Taylor')} />);
