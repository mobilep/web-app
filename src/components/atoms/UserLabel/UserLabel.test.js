import React from 'react';
import { shallow } from 'enzyme';

import UserLabel from './UserLabel';

describe('Atom - UserLabel', () => {

	it('Should render', () => {
		const wrapper = shallow(
			<UserLabel
				fullName='Xena Warrior-Princess'
				initials='XW'
				imgUrl='url'
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});

});
