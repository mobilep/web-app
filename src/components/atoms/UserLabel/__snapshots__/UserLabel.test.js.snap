// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom - UserLabel Should render 1`] = `
<div
  className="UserLabel"
>
  <Avatar
    border={false}
    className="UserLabel-avatar"
    img="url"
    initials="XW"
    size="avatar-22"
  />
  Xena Warrior-Princess
</div>
`;
